src/spatial-index.ts

```diff
@@ -0,0 +1,125 @@
+import {
+  DEFAULT_AGGREGATION_RES_LEVEL_H3,
+  DEFAULT_AGGREGATION_RES_LEVEL_QUADBIN,
+} from './constants-internal';
+import {ModelSource} from './models/model';
+import {AggregationOptions} from './sources/types';
+
+const DEFAULT_TILE_SIZE = 512;
+const QUADBIN_ZOOM_MAX_OFFSET = 4;
+
+export function getSpatialFiltersResolution({
+  source,
+  viewState,
+}: {
+  source: ModelSource & AggregationOptions;
+  viewState: {
+    zoom: number;
+    latitude: number;
+    longitude: number;
+  };
+}) {
+  if (source.spatialDataType === 'geo') {
+    return undefined;
+  }
+
+  const currentZoom = viewState.zoom ?? 1;
```

Comment:
Minor — Since `viewState.zoom` is required input, I'd slightly prefer not giving it a fallback here.

Reply:
Ok we will remove this variable then